### PR TITLE
Fix and catch speech computation errors, add missing braille option

### PR DESCRIPTION
**Pull request by zorkow**
PR contains some corrections.

* Builds the correct speech string from the SSML also for the `mjx-container` element's `aria-label`.
* No speech is computed when `enableEnrichment` is set to `false`. Also catches error on speech computation when `enableEnrichment` is `true` but the element is actually not enriched. I think this currently only happens in the lab or when an enrichment error is caught.
* Adds missing `enableBraille` option.

### CODE REVIEW

File: ts/a11y/semantic-enrich.ts [hunk 1]
```diff
@@ -244,12 +244,21 @@ export function EnrichedMathItemMixin<N, T, D, B extends Constructor<AbstractMat
      */
     public attachSpeech(document: MathDocument<N, T, D>) {
       if (this.state() >= STATE.ATTACHSPEECH) return;
+      if (this.isEscaped || !document.options.enableEnrichment) {
+        this.state(STATE.ATTACHSPEECH);
```
Comment (dpvc) [inline ts/a11y/semantic-enrich.ts:248]: Do any of the calls below cause a MathJax retry error?  (That is, do any call `mathjax.retryAfter()`?)

If not, then you can do `this.state(STATE.ATTACHSPEECH)` above line 247, and then just return here and the other places where the state is set.  That would avoid having to set the state in several places.  But it only works if the rest of the code will always complete and not restart the typesetting via `retryAfter()`.

File: ts/a11y/semantic-enrich.ts [hunk 2]
```diff
@@ -244,12 +244,21 @@ export function EnrichedMathItemMixin<N, T, D, B extends Constructor<AbstractMat
...
+        return;
+      }
       let [speech, braille] = this.existingSpeech();
       let [newSpeech, newBraille] = ['', ''];
       if (!speech || !braille ||
         document.options.enableSpeech || document.options.enableBraille) {
```
Comment (dpvc) [inline ts/a11y/semantic-enrich.ts:254]: I didn't catch this earlier, but I think this condition is wrong.  That is, if `document.options.enableSpeech` is false, you should not generate speech even if `!speech` is true, and similarly for braille.  So I think this should be

``` js
if ((!speech && document.options.enableSpeech) || (!bralle && document.options.enableBraille)) {
```

**Review by dpvc — CHANGES_REQUESTED** [review]
There are a couple of items identified here for your consideration.

Also, it looks like the function 

https://github.com/mathjax/MathJax-src/blob/799ad89bcfc2dd750c7e970214d87065117a4c01/ts/a11y/semantic-enrich.ts#L290-L306

is no longer used, so can be eliminated.

Also, I'm wondering if

``` js
if (this.options.enableSpeech || this.options.enableBraille) {
...
}
```

should be placed around lines

https://github.com/mathjax/MathJax-src/blob/799ad89bcfc2dd750c7e970214d87065117a4c01/ts/a11y/semantic-enrich.ts#L417-L419

like is done at line 430 below.  That would save having to loop through all the math when we aren't adding speech or braille.

File: ts/a11y/semantic-enrich.ts (hunk 1 above)
Comment (zorkow) [reply ts/a11y/semantic-enrich.ts:248]: It should not. The only place that triggers a `retryAfter` is in line 183:
https://github.com/mathjax/MathJax-src/blob/799ad89bcfc2dd750c7e970214d87065117a4c01/ts/a11y/semantic-enrich.ts#L183
But that should have run before `attachSpeech` at all times.

File: ts/a11y/semantic-enrich.ts (hunk 2 above)
Comment (zorkow) [reply ts/a11y/semantic-enrich.ts:254]: Thanks. Good catch.

**Comment by zorkow** [comment]
I've made changes except the `attachSpeech` one and I've made a change in `package.json` to the new SRE release.

**Review by dpvc — APPROVED** [review]
OK, this all looks good.  If you want to do the `STATE.ATTACHSPEECH` change, that is great.  Approved either way.

**Comment by zorkow** [comment]
> OK, this all looks good. If you want to do the `STATE.ATTACHSPEECH` change, that is great. Approved either way.

I've tested with the states and I've made the simplification.
